src/api/transactions.coffee

```diff
@@ -220,12 +228,71 @@ exports.updateTransaction = (transactionId) ->
 
   transactionId = unescape transactionId
   updates = this.request.body
+  that = this
 
   try
-    yield transactions.Transaction.findByIdAndUpdate(transactionId, updates).exec()
-    this.body = "Transaction with ID: #{transactionId} successfully updated"
-    this.status = 'ok'
-    logger.info "User #{this.authenticated.email} updated transaction with id #{transactionId}"
+    yield transactions.Transaction.findByIdAndUpdate transactionId, updates, () ->
+      that.body = "Transaction with ID: #{transactionId} successfully updated"
+      that.status = 'ok'
+      logger.info "User #{that.authenticated.email} updated transaction with id #{transactionId}"
+
+      ###
+      # Update transaction metrics
+      ###
+      transactions.Transaction.findById transactionId, (err, doc) ->
+        if updates['$push'].routes?
```

Comment:
You have to set the push property so that you don't overwrite the other secondary routes with your update.